docs/getting-started/3-creating-tasks.md

```diff
@@ -0,0 +1,191 @@
+// It can also be used within the `series()` composition.
+function build(cb) {
+  // body omitted
+  cb();
+}
+
+exports.build = build;
+exports.default = series(clean, build);
+```
+
+![ALT TEXT MISSING][img-gulp-tasks-command]
+
+<small>In the past, `task()` was used to register your functions as tasks. While that API is still available, exporting should be the primary registration mechanism, except in edge cases where exports won't work.</small>
+
+## Compose tasks
+
+Gulp provides two powerful composition methods, `series()` and `parallel()`, allowing individual tasks to be composed into larger operations. Both methods accept any number of task functions or composed operations.  `series()` and `parallel()` can be nested within themselves or each other to any depth.
+
+To have your tasks execute in order, use the `series()` method.
+```js
+const { series } = require('gulp');
+
+function transpile(cb) {
+  // body omitted
+  cb();
+}
+
+function bundle(cb) {
+  // body omitted
+  cb();
+}
+
+exports.build = series(transpile, bundle);
+```
+
+For tasks to run at maximum concurrency, combine them with the `parallel()` method.
+```js
+const { parallel } = require('gulp');
+
+function javascript(cb) {
+  // body omitted
+  cb();
+}
+
+function css(cb) {
+  // body omitted
+  cb();
+}
+
+exports.build = parallel(javascript, css);
+```
+
+Tasks are composed immediately when either `series()` or `parallel()` is called.  This allows variation in the composition instead of conditional behavior inside individual tasks.
+
+```js
+const { series } = require('gulp');
+
+function minify(cb) {
+  // body omitted
+  cb();
+}
+
+
+function transpile(cb) {
+  // body omitted
+  cb();
+}
+
+function livereload(cb) {
+  // body omitted
+  cb();
+}
+
+if (process.env.NODE_ENV === 'production') {
+  exports.build = series(transpile, minify);
+} else {
+  exports.build = series(transpile, livereload);
+}
+```
+
+`series()` and `parallel()` can be nested to any arbitrary depth.
+
+```js
+const { series, parallel } = require('gulp');
+
+function clean(cb) {
+  // body omitted
+  cb();
+}
+
+function cssTranspile(cb) {
+  // body omitted
+  cb();
+}
+
+function cssMinify(cb) {
+  // body omitted
+  cb();
+}
+
+function jsTranspile(cb) {
+  // body omitted
+  cb();
+}
+
+function jsBundle(cb) {
+  // body omitted
+  cb();
+}
+
+function jsMinify(cb) {
+  // body omitted
+  cb();
+}
+
+function publish(cb) {
+  // body omitted
+  cb();
+}
+
+exports.build = series(
+  clean,
+  parallel(
+    cssTranspile,
+    series(jsTranspile, jsBundle)
+  ),
+  parallel(cssMinify, jsMinify),
+  publish
+);
+```
+
+When a composed operation is run, each task will be executed every time it was referenced.  For example, a `clean` task referenced before two different tasks would be run twice and lead to undesired results.  Tasks can be wrapped with the [async-once][async-once] module if this **(not recommended)** pattern is needed.
```

Comment:
Is there a recommended pattern for the same problem? Otherwise I think this note isn't very helpful.

Reply:
This is a pattern that people migrating from 3.x might encounter so I thought it was worth mentioning that they can use `async-once` but the "right way" to do it would be to restructure your tasks so the task was only specified once - the pattern recommended throughout the document.

I'm open to adjusting the language but I think this is worthwhile enough to keep.